crates/ruff_python_formatter/src/expression/expr_await.rs

```diff
@@ -15,12 +15,48 @@ impl FormatNodeRule<ExprAwait> for FormatExprAwait {
     fn fmt_fields(&self, item: &ExprAwait, f: &mut PyFormatter) -> FormatResult<()> {
         let ExprAwait { range: _, value } = item;
 
+        // Drop parentheses around low-precedence operators (constants, names, attributes, and
+        // subscripts). It would also be syntactically valid to drop parentheses around lists,
+        // sets, and dictionaries, but Black doesn't do that.
+        // See: https://docs.python.org/3/reference/expressions.html#operator-precedence
+        let parenthesize = match value.as_ref() {
```

Comment:
I see how this is somewhat easier to implement but wonder it it should instead be moved into NeedsParentheses of the individual expressions to better align with the overall design

Reply:
Hmm... I'm torn? This feels so much easier to understand since it's in a centralized place that's coupled to the `await` formatting, but I can understand why moving this into the individual expressions adheres more closely to the overall design.

Reply:
On the other hand, this feels more consistent with how (e.g.) `expr_yield` works? We always preserve parentheses (`Parenthesize::Optional`), and so don't require that expressions encode yield precedence in their own `NeedsParentheses`.

Reply:
> On the other hand, this feels more consistent with how (e.g.) `expr_yield` works? We always preserve parentheses (`Parenthesize::Optional`), and so don't require that expressions encode yield precedence in their own `NeedsParentheses`.

Isn't that different because `yield` always preserve parentheses for all kind of expressions whereas `await` only preserves for some (which is what `NeedsParentheses` is for).

Reply:
Yes but isn't it strange that the `NeedsParentheses` for (e.g.) boolean operators would need to take into account whether we always preserve parentheses for `yield` vs. only sometimes do so for `await`? Parentheses are required in both cases, but the suggestion here is that we add a condition for `await` to the `NeedsParentheses` implementation.

(Perhaps the "correct" solution would be to include both `await` and `yield` and whatever else in that `NeedsParentheses` implementation, even though it would be redundant for `yield` in practice due to the fact that we preserve parentheses in the parent.)

Reply:
Maybe `Yield` should be changed too ;) Altough I think `yield` is different because the parentheses are always preserves (as far as I understand). 

This is different from `await` where the use is a "hack" to get the `Always` because it simply relies on the fact that the parentheses are already there. This is also slightly slower because the formatter has to test whether the expression is parenthesized in the source. 

We could include `yield` in `NeedsParentheses` but it ends up being untested code.